Switching from Netbeans makefile to a global cmake

Sunseille opened the pull request:
Working on a cmake file that will let panel-pop being IDE dependant, and let build on different OS.
It's not finished and i can't test at all on windows or MacOS since i'm not a user of theses OS.

The major change was the deletion of nbproject folder.
I added a CMake folder including cmake scripts who'll search if SDL is installed on the computer (working on OSX?)

I also tried to add changes from: https://github.com/a544jh/panel-pop/pull/4 since it hasn't been merged and OSX support will be esier to implement with cmake.

Commit:
(WIP) Switching from Netbeans makefile to a global cmake

Working on a cmake file that will let panel-pop being IDE dependant, and let build on different OS.
It's not finished and i can't test at all on windows or MacOS since i'm not a user of theses OS.

The major change was the deletion of nbproject folder.
I added a CMake folder including cmake scripts who'll search if SDL is installed on the computer (working on OSX?)

I also tried to add changes from: https://github.com/a544jh/panel-pop/pull/4 since it hasn't been merged and OSX support will be esier to implement with cmake.

Commit:
typo correction

Commit:
Update README.md

Commit:
Update README.md

ghost commented:
updated:
 -forgot to delete a tmp file
 -changed README.md (a lot) because i'm not good with markdown syntax

a544jh commented:
Got it to work on Windows
* have MinGW and cmake in your PATH
* change `if(LINUX)` to `if(LINUX OR WIN32)` in CMakeLists.txt, line 75
* run `cmake -G "MinGW Makefiles" -DCMAKE_SH="CMAKE_SH-NOTFOUND" .`
* run `mingw32-make.exe`

(I don't remember how I installed Boost anymore :P)

Commit:
Update README.md

Commit:
removed if statement

removed if statement for OS support. He doesn't seems we need it right now so there's no reason to keep it.

also updated README.md to add a quick building guide

Commit:
.gitignore updated

updated for the cmake compilation

Commit:
CMakeLists.txt should not be in gitignore
